sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ForLoopIncrementSignCheck.cs

```diff
@@ -34,25 +34,206 @@ namespace SonarAnalyzer.Rules.CSharp
+                case SyntaxKind.AddAssignmentExpression:
+                    if (incrementor is AssignmentExpressionSyntax add && add.Left is IdentifierNameSyntax leftId &&
+                        add.Right.IsKind(SyntaxKind.NumericLiteralExpression))
+                    {
+                        identifierName = leftId.Identifier.ValueText;
+                        opp = incrementorToOperation.GetValueOrDefault(incrementorKind, ArithmeticOperation.None);
+                    }
+                    break;
+
+                case SyntaxKind.SimpleAssignmentExpression:
+                    if (incrementor is AssignmentExpressionSyntax simpleAssignment &&
+                        simpleAssignment.Left is IdentifierNameSyntax simpleAssignmentId &&
+                        simpleAssignment.Right is BinaryExpressionSyntax right &&
+                        IsVariableAndLiteralBinaryExpression(right, simpleAssignmentId.Identifier.ValueText))
+                    {
+                        identifierName = simpleAssignmentId.Identifier.ValueText;
+                        opp = incrementorToOperation.GetValueOrDefault(right.Kind(), ArithmeticOperation.None);
+                    }
+                    break;
+
+                default:
+                    break;
+            }
+
+            return new IncrementData(identifierName, opp);
+        }
+
+        private Condition GetCondition(BinaryExpressionSyntax conditionSyntax, string identifierName)
+        {
+            if (conditionSyntax.Left is IdentifierNameSyntax leftIdentifier && leftIdentifier.NameIs(identifierName))
+            {
+                if (conditionSyntax.IsAnyKind(SyntaxKind.LessThanExpression, SyntaxKind.LessThanOrEqualExpression))
+                {
+                    return Condition.Less;
+                }
+                else if (conditionSyntax.IsAnyKind(SyntaxKind.GreaterThanExpression, SyntaxKind.GreaterThanOrEqualExpression))
+                {
+                    return Condition.Greater;
+                }
+            }
+            else if (conditionSyntax.Right is IdentifierNameSyntax rightIdentifier && rightIdentifier.NameIs(identifierName))
+            {
+                if (conditionSyntax.IsAnyKind(SyntaxKind.LessThanExpression, SyntaxKind.LessThanOrEqualExpression))
+                {
+                    return Condition.Greater;
+                }
+                else if (conditionSyntax.IsAnyKind(SyntaxKind.GreaterThanExpression, SyntaxKind.GreaterThanOrEqualExpression))
+                {
+                    return Condition.Less;
+                }
+            }
+
+            return Condition.None;
+        }
+
+        private bool IsVariableAndLiteralBinaryExpression(BinaryExpressionSyntax binaryExpression, string identifier) =>
+            (binaryExpression.Left is IdentifierNameSyntax leftId && leftId.NameIs(identifier) && binaryExpression.Right.IsKind(SyntaxKind.NumericLiteralExpression)) ||
```

Comment:
having a simple method `IsIdentifier(SyntaxNode x, string identifier)` would make this method more readable

Reply:
Done. Simplified a bit also **GetCondition** with this.

Reply:
nice 😄